hashintel/hash
# H-716, H-826, H-828, H-873, H-926: Give system types hash.ai URL, remove system user

## 🌟 What is the purpose of this PR?

The main purpose of this PR is to make system types have a fixed `https://hash.ai` URL (H-828). 

"System types" are those relied on by the HASH application, e.g. `User`, `Page`, and which must be available in any HASH instance for it to function. Previously, system types would be seeded on instance startup under a 'system user' namespace  defined by an environment variable, using the origin of the frontend for that instance, meaning every instance hosted its own copy of these types under a different URL and namespace.

This PR changes that so that all system types are hosted at `https://hash.ai` under a `hash` namespace, in anticipation of the hosted HASH instance being hosted at that origin. Other instances of HASH will load system types during seeding as 'external types', i.e. types which do not belong to a web/namespace in the instance. The main motivation for this is that these are all in fact the same type, so they should have the same id (their full URL).

It also makes all types which are created while running on https://app.hash.ai have a https://hash.ai URL, in anticipation of the app being migrated there.




## 🔍 What does this change?




- Update the seeding script for system types to give them a `https://hash.ai` URL (H-828), and load them as external types for self-hosted instances (H-716). As part of this it ensures they are in the correct webs/namespaces (previously the Linear types were not in the `linear` web)
- As a temporary measure while hosted HASH is not in fact at https://hash.ai, allow the frontend hosted at https://app.hash.ai to serve types with a https://hash.ai URL – to be removed once the app is migrated to https://hash.ai, in H-1172. 
- Remove the system user (H-926)
- Remove the admin link – admins have for a little while been defined by an account group, not this link (H-826)
- Show something in the types table's 'Web' column for external types (previously 'Namespace') (H-873)
- Fix an issue where property types would sometimes fail to load (if a user logged in then visited the types page), leaving the types page to hang loading indefinitely – the issue was authentication being required to fetch the property types, which is not necessary and I've removed
- Rename the `types` object to `systemTypes` for easier discoverability

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## 🐾 Next steps

1. The production db will need wiping when this is merged
2. Regenerate the system type TypeScript types once this is deployed on https://app.hash.ai – H-1171

## 🛡 What tests cover this?



- Some existing

## ❓ How to test this?



1. Locally, wipe the db
2. Run `yarn dev` so that the API seeds the new system types
3. Visit the types page to see the created types
4. You can add a path to a system type to `http://localhost:3000/` to visit it (e.g. `http://localhost:3000/@hash/types/entity-type/org/v/1`)

## 📹 Demo



https://github.com/hashintel/hash/assets/37743469/f6573d05-a81b-42ce-8e39-16b3547f6cbf
